Terminal.Gui/ViewBase/Layout/SuperViewChangingEventArgs.cs

```diff
@@ -0,0 +1,27 @@
+using Terminal.Gui.App;
+
+namespace Terminal.Gui.ViewBase;
+
+/// <summary>
+///     Args for events where the <see cref="View.SuperView"/> of a <see cref="View"/> is about to be changed (e.g.
+///     before <see cref="View.Remove(View?)"/>).
+/// </summary>
+/// <remarks>
+///     <para>
+///         This event is raised before the <see cref="View.SuperView"/> property is changed, allowing access to the
+///         current SuperView and its associated resources (such as <see cref="View.App"/>) for cleanup purposes.
+///     </para>
+///     <para>
+///         This event follows the Cancellable Work Pattern (CWP) and can be cancelled by setting <see cref="System.ComponentModel.CancelEventArgs.Cancel"/> to <see langword="true"/>.
+///     </para>
+/// </remarks>
+public class SuperViewChangingEventArgs : CancelEventArgs<View?>
```

Comment:
This class is not needed. Just use CancelEventArgs<View?> directly. No need to convey old value (subscriber can just access property). Having these bespoke event types is legacy complexity and we're trying to clean that up.